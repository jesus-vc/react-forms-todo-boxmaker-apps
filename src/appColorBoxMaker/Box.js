import "./Box.css";

const Box = ({ removeBox, backgroundColor, width, height }) => {
  return (
    <>
      <div
        className="Box"
        data-testid="singleBox"
        style={{
          backgroundColor: backgroundColor,
          width: width,
          height: height,
        }}
      ></div>
      <button className="Box-Button" onClick={removeBox}>
        X
      </button>
    </>
  );
};

export default Box;
