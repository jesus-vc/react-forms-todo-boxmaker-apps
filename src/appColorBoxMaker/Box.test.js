import Box from "./Box";
import { fireEvent, render, screen } from "@testing-library/react";

// smoke test
it("renders without crashing", () => {
  render(<Box />);
});

//snapshot text
it("matches snapshot", () => {
  const { asFragment } = render(<Box></Box>);
  expect(asFragment()).toMatchSnapshot();
});

it("renders a div with props", () => {
  const backgroundColor = "black";
  const width = "50px";
  const height = "100px";
  render(
    <Box backgroundColor={backgroundColor} width={width} height={height}></Box>
  );
  const box = screen.getByTestId("singleBox");
  expect(box).toHaveStyle({
    backgroundColor: backgroundColor,
    width: width,
    height: height,
  });
});

it("contains a button with the text of 'X'", () => {
  render(<Box></Box>);
  const button = screen.getByRole("button", { name: "X" });
  expect(button).toBeInTheDocument();
});
