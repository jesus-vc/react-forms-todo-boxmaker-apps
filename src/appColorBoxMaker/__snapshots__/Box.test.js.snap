// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="singleBox"
  />
  <button
    class="Box-Button"
  >
    X
  </button>
</DocumentFragment>
`;
